### ISSUE TITLE
LOD Syncing -  A 'Synced' status is displayed instead of 'Not enough storage' when there is a storage limitation on the LOD

### ISSUE THREAD

**Issue by pcenov** [learningequality/kolibri#11505]
## Observed behavior
When a Coach is checking the status of the connected devices and there is an imposed storage limit on the LOD then I see the 'Synced' status instead of the 'Not enough storage' status. So far I have confirmed this behavior on Ubuntu server and Ubuntu LOD, will further test it tomorrow on Android. When I further tried testing the behavior with the download limit set to 1 GB the syncing stopped completely.

## Expected behavior
The correct status should be displayed in that case.

## Steps to reproduce the issue
1. Install the [Beta 7 build] from this PR (https://github.com/learningequality/kolibri/pull/11489).
2. Setup a server and a LOD device connected to the server.
3. As the super admin of the LOD device go to  Device > Settings and set the storage limit to 0
4. As a coach go to Coach > Plan > Lessons, create a new lesson and make it visible
5. Observe that on the LOD device the status gets changed to 'Not enough storage' while at Coach > Class home > View learners it's displayed as 'Synced'

## Screenshot

![2023-11-07_16-58-55](https://github.com/learningequality/kolibri/assets/79847249/8dceed50-8e56-4fed-bc87-bc956f58f841)

## Logs

https://drive.google.com/drive/folders/1jRD_nHghaxbwgsloXqhqGnomb4ONJQlH?usp=drive_link

## Usage Details
Ubuntu 22  - Chrome, Firefox

**Comment by pcenov** [comment]
@radinamatic

**Comment by pcenov** [comment]
In addition to the above mentioned scenario, today I imported a learner on an Android device, and then signed in as the same learner set the storage limit to 1GB and stopped the app. Then on the Ubuntu server I enabled several lessons with a total size greater than 1GB. When I started the app again it immediately tried to download the lessons, the banner for not enough storage got displayed and the sync status was 'Not enough storage' it proceeded attempting to download all resources and then at some point it just deleted all the channels and the status changed to 'Synced' as if nothing had happened.

I was able to replicate the same behavior with Ubuntu LOD.

Shorter Ubuntu LOD video:

https://github.com/learningequality/kolibri/assets/79847249/04ad1415-a3d3-41c2-ba5f-7a6cd4c8e267

Android LOD video (shortened to 5 minutes instead of the original 10):

https://github.com/learningequality/kolibri/assets/79847249/f1d6fda3-862b-4823-b8f5-e7013ba2c803

Logs: https://drive.google.com/drive/folders/1KK-F_omgXHIuoRszFpMvuA0WWqyCQmvz?usp=drive_link
